language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ClientResourceAccessActionNodeContext.java

```diff
@@ -77,24 +96,158 @@ public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context,
+        if (segments.size() < currentSegments.size()) {
+            return Pair.of(Collections.emptyList(), false);
+        }
+
+        if (currentSegments.isEmpty() && ResourcePathCompletionUtil.isInMethodCallContext(accNode, context)) {
+            return Pair.of(Collections.emptyList(),
+                    resourceMethodSymbol.resourcePath().kind() == ResourcePath.Kind.DOT_RESOURCE_PATH);
+        }
+
+        int completableSegmentStartIndex = 0;
+        int numOfProvidedSegments = currentSegments.size();
+        for (int i = 0; i < numOfProvidedSegments; i++) {
+            completableSegmentStartIndex += 1;
+            Node node = currentSegments.get(i);
+            PathSegment segment = segments.get(i);
+            if (node.kind() == SyntaxKind.IDENTIFIER_TOKEN
+                    && segment.pathSegmentKind() == PathSegment.Kind.NAMED_SEGMENT) {
+                String currentPath = ((IdentifierToken) node).text().strip();
+                String expectedPath = ((NamedPathSegment) segment).name();
+                if (i < numOfProvidedSegments - 1 && currentPath.equals(expectedPath)) {
+                    continue;
+                } else if (i == numOfProvidedSegments - 1) {
+                    if (currentPath.equals(expectedPath)) {
+                        if (context.getCursorPositionInTree() == node.textRange().endOffset()) {
+                            //covers /path1<cursor>, excluding /path1/<cursor> and /path1.<cursor>
+                            completableSegmentStartIndex -= 1;
+                        }
+                        continue;
+                    } else if (expectedPath.startsWith(currentPath)
+                            && context.getCursorPositionInTree() == node.textRange().endOffset()) {
+                        //suggest only when /pat<cursor> excluding /pat/<cursor>
+                        completableSegmentStartIndex -= 1;
+                        continue;
+                    }
+                }
+                //Covers named segments that are not matching
+                return Pair.of(Collections.emptyList(), false);
+            } else if (node.kind() == SyntaxKind.COMPUTED_RESOURCE_ACCESS_SEGMENT
+                    && (segment.pathSegmentKind() == PathSegment.Kind.PATH_PARAMETER ||
+                    segment.pathSegmentKind() == PathSegment.Kind.PATH_REST_PARAMETER)) {
+                TypeSymbol typeSymbol = segment.pathSegmentKind() == PathSegment.Kind.PATH_REST_PARAMETER ?
+                        ((ArrayTypeSymbol) (((PathParameterSymbol) segment).typeDescriptor())).memberTypeDescriptor() :
+                        ((PathParameterSymbol) segment).typeDescriptor();
+                Optional<SemanticModel> semanticModel = context.currentSemanticModel();
+                if (semanticModel.isEmpty()) {
+                    return Pair.of(Collections.emptyList(), false);
```

Comment:
There are some code lines that are not covered by test cases are those false positives or expected?

Reply:
Will check and update

Reply:
Added a test case covering one of the cases that was not covered. Other cases that are reported can be ignored